pallets/xcm-executor-utils/src/tests.rs

```diff
@@ -0,0 +1,145 @@
+#[test]
+fn teleport_policy_can_be_set_and_removed() {
+    new_test_ext().execute_with(|| {
+        let origin_multilocation = MultiLocation::parent();
+        let trust_policy = TrustPolicy::DefaultTrustPolicy(DefaultTrustPolicy::Never);
+
+        let _ = XcmExecutorUtils::set_teleport_policy(
+            RuntimeOrigin::root(),
+            origin_multilocation,
+            trust_policy.clone(),
+        );
+
+        assert_eq!(
+            XcmExecutorUtils::teleport_policy(origin_multilocation),
+            Some(trust_policy)
+        );
+
+        let _ =
+            XcmExecutorUtils::remove_teleport_policy(RuntimeOrigin::root(), origin_multilocation);
+
+        assert!(XcmExecutorUtils::teleport_policy(origin_multilocation).is_none());
+    });
+}
+
+#[test]
+fn reserve_policy_is_applied() {
+    new_test_ext().execute_with(|| {
+        let parent_multilocation = MultiLocation::parent();
+
+        let parent_asset = MultiAsset {
+            id: AssetId::Concrete(MultiLocation::parent()),
+            fun: Fungible(1_000),
+        };
+
+        let grandparent_asset = MultiAsset {
+            id: AssetId::Concrete(MultiLocation::grandparent()),
+            fun: Fungible(1_000),
+        };
+
+        // Only allow grandparent_asset
+        let _ = XcmExecutorUtils::set_reserve_policy(
+            RuntimeOrigin::root(),
+            parent_multilocation,
+            TrustPolicy::AllowedAssets(BoundedVec::try_from(vec![grandparent_asset.id]).unwrap()),
+        );
+
+        // Should allow grandparent_asset
+        assert!(filters::IsReserveFilter::<Test>::contains(
+            &grandparent_asset,
+            &parent_multilocation
+        ));
+
+        // Should reject parent_asset
+        assert_eq!(
+            IsReserveFilter::<Test>::contains(&parent_asset, &parent_multilocation),
+            false
+        );
+    });
+}
+
+#[test]
+fn teleport_policy_is_applied() {
+    new_test_ext().execute_with(|| {
+        let parent_multilocation = MultiLocation::parent();
+
+        let parent_asset = MultiAsset {
+            id: AssetId::Concrete(MultiLocation::parent()),
+            fun: Fungible(1_000),
+        };
+
+        let grandparent_asset = MultiAsset {
+            id: AssetId::Concrete(MultiLocation::grandparent()),
+            fun: Fungible(1_000),
+        };
+
+        // Only allow grandparent_asset
+        let _ = XcmExecutorUtils::set_teleport_policy(
+            RuntimeOrigin::root(),
+            parent_multilocation,
+            TrustPolicy::AllowedAssets(BoundedVec::try_from(vec![grandparent_asset.id]).unwrap()),
+        );
+
+        // Should allow grandparent_asset
+        assert!(IsTeleportFilter::<Test>::contains(
+            &grandparent_asset,
+            &parent_multilocation
+        ),);
+
+        // Should reject parent_asset
+        assert_eq!(
+            IsTeleportFilter::<Test>::contains(&parent_asset, &parent_multilocation),
+            false
+        );
+    });
+}
```

Comment:
I would add a test testing the default policy

Reply:
Added some tests for default policy https://github.com/moondance-labs/dancekit/pull/5/files#diff-232cefac2a0edaa4c37c6567b57bf0387c00398866991963bf49fbdb9854f27fR144-R212